pkg/koordlet/metricsadvisor/collectors/nodeinfo/node_info_collector.go

```diff
@@ -86,3 +87,30 @@ func (n *nodeInfoCollector) collectNodeCPUInfo() {
 	klog.Infof("collectNodeCPUInfo finished, cpu info: processors %v", len(nodeCPUInfo.ProcessorInfos))
 	metrics.RecordCollectNodeCPUInfoStatus(nil)
 }
+
+func (n *nodeInfoCollector) collectNodeLocalStorageInfo() {
+	klog.V(6).Info("start collect node local storage info")
+
+	localStorageInfo, err := koordletutil.GetLocalStorageInfo()
+	if err != nil {
+		klog.Warningf("failed to collect node local storage info, err: %s", err)
+		metrics.RecordCollectNodeLocalStorageInfoStatus(err)
+		return
+	}
+
+	nodeLocalStorageInfo := &metriccache.NodeLocalStorageInfo{}
+	nodeLocalStorageInfo.DiskNumberMap = localStorageInfo.DiskNumberMap
+	nodeLocalStorageInfo.NumberDiskMap = localStorageInfo.NumberDiskMap
+	nodeLocalStorageInfo.PartitionDiskMap = localStorageInfo.PartitionDiskMap
+	nodeLocalStorageInfo.VGDiskMap = localStorageInfo.VGDiskMap
+	nodeLocalStorageInfo.LVMapperVGMap = localStorageInfo.LVMapperVGMap
+	nodeLocalStorageInfo.MPDiskMap = localStorageInfo.MPDiskMap
+
+	klog.V(6).Infof("collect node local storage info finished, nodeCPUInfo %v", localStorageInfo)
+	if err = n.metricDB.InsertNodeLocalStorageInfo(nodeLocalStorageInfo); err != nil {
+		klog.Errorf("insert node local storage info error: %v", err)
+	}
+
+	n.started.Store(true)
```

Comment:
Some other modules rely on the status `started` of the CPU Info while sharing the status in `collectNodeLocalStorageInfo` may corrupt the condition.  
e.g. The nodeTopologyInformer requires the CPU Info synced, so it can report the correct CPU topology.
How about moving the collection of the local storage into a new plugin?

Reply:
Maybe I can rename `NodeInfoCollector` plugin to `NodeCPUInfoCollector` plugin, and add a new plugin called `NodeStorageInfoCollector`?

Reply:
Or just remove `n.started.Store(true)`?

Reply:
Yes, I prefer option one.

Reply:
done